Gộp welcome screen & login func vào branch test

Huynguyen736 opened the pull request.

Huynguyen736 committed:
Done Welcome and login UI

Huynguyen736 committed:
-refactor code-

cuongceg committed:
fix: change location of file component

cuongceg committed:
fix: fix location path

Huynguyen736 commented:
hẹ hẹ hẹ

Huynguyen736 commented:
@cuongceg a cuong cuu e

cuongceg reviewed (commented):
Không change cho iOS à em

cuongceg commented on lib/core/theme/app_fonts.dart:
```diff
@@ -0,0 +1,189 @@
+  AppFonts._();
+
+  // Font family
+  static const String primaryFont = 'Alexandria';
```
không dùng biến này thì xoá đi nhé

cuongceg commented on lib/core/theme/app_theme.dart:
```diff
@@ -0,0 +1,119 @@
+      brightness: Brightness.dark,
+      primaryColor: AppColors.primary,
+      scaffoldBackgroundColor: AppColors.backgroundDark,
+      // Add dark theme configuration here when needed
```
add nốt đi ní :)))

cuongceg commented on lib/features/login/ui/login_screen.dart:
```diff
@@ -0,0 +1,212 @@
+                        width: screenWidth * 0.15,
+                        height: screenWidth * 0.15,
+                        child: ClipOval(
+                          child: Image.asset(AppImages.googleIcon, fit: BoxFit.cover),
```
lừa quá :))).Không phải icon button à

cuongceg commented on lib/features/login/ui/login_screen.dart:
```diff
@@ -0,0 +1,212 @@
+    return Scaffold(
+      backgroundColor: AppColors.backgroundLight,
+      body: SafeArea(
+        child: Center(
```
bỏ thẻ center này đi cũng được dùng horrizontal: center nhé

cuongceg commented on lib/features/login/ui/login_screen.dart:
màn này UI ổn nhưng UX chưa ổn khi anh dùng điện thoại to lắm rồi mà phải vuốt xuống thấy nút đăng ký 
=> cân đối lại khoảng cách
<img width="392" height="838" alt="Ảnh màn hình 2025-07-24 lúc 22 54 10" src="https://github.com/user-attachments/assets/f721c742-84e9-4de2-a89e-072283d130b6" />

cuongceg commented on lib/features/login/ui/signup_popup.dart:
Responsive cho riêng ra một file nhé để còn tái sử dụng

cuongceg commented on lib/features/login/ui/signup_popup.dart:
```diff
@@ -0,0 +1,257 @@
+    );
+  }
+
+  Widget _buildFormField({
```
ý sao không dùng cái này trong màn Log in em 👍

Huynguyen736 replied:
cái lõi của hàm này thì nó cũng là hàm ở trong login mà a, mở rông thêm thoii

cuongceg commented on lib/features/login/ui/splash_screen.dart:
không dùng thì xoá đi cha 👍

cuongceg commented on lib/main.dart:
```diff
@@ -1,3 +1,36 @@
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  SharedPreferences prefs = await SharedPreferences.getInstance();
```
cái này viết ra file riêng của shared preference nhé, sau này nhiều cái khác cũng dùng. Thêm vào utils nhé

cuongceg commented:
@Huynguyen736 review sơ sơ thế nhé :)))

Huynguyen736 committed:
feat: Update application icon on Android and iOS

Huynguyen736 committed:
feat: add dark theme config and fix icon app

sclearner commented on lib/features/login/ui/login_screen.dart:
```diff
@@ -0,0 +1,212 @@
+                            obscureText: _isPasswordObscured,
+                            suffixIcon: IconButton(
+                              icon: Icon(
+                                _isPasswordObscured ? Icons.visibility_off : Icons.visibility,
```
Icon mắt bị ngược rồi nè. UI của button thể hiện hành động nên là khi ẩn hiện mắt mở còn khi hiện mật khẩu thì hiện mắt đóng nha

sclearner commented on lib/features/login/ui/signup_popup.dart:
```diff
@@ -0,0 +1,257 @@
+  );
+}
+
+class _SignUpDialogContent extends StatelessWidget {
```
Dùng design pattern Abstract Factory + check platform + LayoutBuilder đi. Adaptive này cấu trúc hơi mù mắt :')

sclearner commented on lib/features/login/ui/signup_popup.dart:
```diff
@@ -0,0 +1,257 @@
+    );
+  }
+
+  Widget _buildFormFields({
```
Hạn chế Widget Builder ntn, chuyển sang class Widget nếu có thể

sclearner commented on lib/features/login/ui/welcome_screen.dart:
```diff
@@ -0,0 +1,212 @@
+                          curve: Curves.easeInOut,
+                        );
+                      } else {
+                        Navigator.pushReplacement(
```
go_router đc ko

sclearner reviewed (changes requested):
Anh bắt được mấy lỗi responsive và UI cơ vả lại đang sai UI đúng không? Với cả hình như em đang đặt kích thước theo % cỡ màn hình. Nếu muốn làm theo hướng adaptive riêng cho từng cỡ màn thì nên dùng design pattern Abstract Factory hoặc Factory Method nhé.
Về cách responsive, 1 là constraints với cha, tạo padding với cha, 2 là giữ nguyên kích thước nhé. Để % dễ bị sai thiết kế á.

Huynguyen736 committed:
chore: Refactor codebase and add improvements
- Implement dark theme configuration.
- Centralize SharedPreferences logic into a dedicated service.
- Extract responsive sizing from SignUpDialog into a reusable class.
- Fix Android build error related to missing mipmap icon.
- Update Google icon widget to a correctly sized IconButton.

Huynguyen736 committed:
chore: login UI responsive for mobile

Huynguyen736 committed:
refactor: Overhaul UI, navigation, and architecture

Implements a comprehensive overhaul based on code review feedback, focusing on creating a robust adaptive/responsive system and migrating to a modern navigation package.

### Adaptive & Responsive UI
- Replaced all percentage-based sizing with a combination of fixed values, layout widgets (`Expanded`, `Padding`), and `lerpDouble` for smooth, controlled scaling of UI elements like buttons and spacing.
- Implemented the **Abstract Factory** pattern (`IWidgetFactory`) to provide platform-specific widgets, ensuring `CupertinoButton` is used on iOS and `ElevatedButton` on other platforms.
- Implemented the **Factory Method** pattern (`LoginLayoutFactory`) to serve entirely different, optimized layouts for mobile and tablet screen sizes.
- Utilized `LayoutBuilder` and `SingleChildScrollView` to enforce a minimum layout width, preventing UI distortion on very narrow screens and enabling horizontal scrolling when necessary.
- Corrected platform-checking logic to be web-compatible by using `kIsWeb` before checking `Platform.isIOS`.

### Navigation
- Migrated from the imperative `Navigator` 1.0 to the declarative `go_router` package.
- Centralized all route definitions and navigation logic into a dedicated `app_router.dart` file.
- Implemented a router-level `redirect` to handle the initial screen logic (Welcome vs. Login) based on shared preferences, cleaning up the `main.dart` file.

### Code Structure
- Refactored large widgets into smaller, single-responsibility files, organizing them into `components`, `layouts`, and `factories` directories.
- Created a shared, reusable `SharedPrefsService` to centralize data storage access.
- Resolved various errors including `LateInitializationError`, incorrect imports, and platform-compatibility issues.

sclearner commented:
<img width="960" height="1020" alt="image" src="https://github.com/user-attachments/assets/b0f01154-02a0-4647-b255-3f4b61086d2e" />
Ôi mắt tôi

sclearner commented on lib/features/login/ui/welcome_screen.dart:
```diff
@@ -1,13 +1,15 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_svg/flutter_svg.dart';
-import 'package:gdgocms/features/login/ui/login_screen.dart';
+import 'package:go_router/go_router.dart';
 import 'package:gdgocms/core/theme/app_colors.dart';
 import 'package:gdgocms/core/theme/app_fonts.dart';
 import 'package:gdgocms/core/theme/app_theme.dart';
 import 'package:gdgocms/core/theme/app_images.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 import 'package:smooth_page_indicator/smooth_page_indicator.dart';
 
+import 'factories/widget_factory.dart';
+
 class WelcomeScreen extends StatefulWidget {
```
<img width="960" height="1020" alt="image" src="https://github.com/user-attachments/assets/9315fca6-01ed-4061-9fc8-94b47e968039" />
=)))

sclearner commented on lib/main.dart:
```diff
@@ -1,36 +1,30 @@
+      routerConfig: AppRouter.router,
     );
   }
 }
```
Animation chuyển trang có vẻ hơi giật giật chậm chậm nha (dù đang chạy trên web). (Thao tác vuốt phải)

sclearner commented on lib/features/login/ui/login_screen.dart:
```diff
@@ -1,11 +1,9 @@
-import 'package:gdgocms/core/theme/app_images.dart';
+import 'package:gdgocms/features/login/ui/factories/login_layout_factory.dart';
 
 class LoginScreen extends StatefulWidget {
```
<img width="960" height="1020" alt="image" src="https://github.com/user-attachments/assets/3c9c78f0-a686-461f-bb74-ae2b0db35a63" />
Mai aiiiiiiizzzzzzz

sclearner commented on lib/features/login/ui/login_screen.dart:
```diff
@@ -20,191 +18,46 @@ class _LoginScreenState extends State<LoginScreen> {
-                          // forgot password
-                          Align(
-                            alignment: Alignment.centerRight,
-                            child: GestureDetector(
```
Inkwell hoặc cho nó gạch chân đi fen. Nút cứng đơ á

sclearner commented on lib/features/login/ui/login_screen.dart:
```diff
@@ -20,191 +18,46 @@ class _LoginScreenState extends State<LoginScreen> {
+          },
         ),
       ),
     );
```
Cái này UX Form có vẻ ok. Ko có điện thoại ko test tiếp mấy cái tap outside lose focus hay chuyển tiếp.

sclearner commented on lib/features/login/ui/signup_popup.dart:
```diff
@@ -170,88 +99,19 @@ class _SignUpDialogContent extends StatelessWidget {
     );
   }
-}
+}
```
Thiếu icon show mật khẩu rồi =)))

sclearner reviewed (changes requested):
Có cải thiện nhưng không đáng kể. Các UI liên quan đến đăng nhập các thứ nên tạo sẵn khái niệm độ trễ cho các nút đi.
